src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0) {
+            initializeTable();
+        }
+        if (getNode(key) != null) {
+            getNode(key).value = value;
+            return;
+        }
+        Node<K, V> newNode = new Node<>(key, value);
+        int index = newNode.hash % capacity;
```

Comment:
you have to implement getIndex(Object key) method to obtain an index value

Reply:
✅